### ISSUE TITLE
Add option to BaseTrainer to freeze entire models or loss functions 

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#92]

**Comment by KevinMusgrave** [comment]
In v0.9.86.dev1, ```freeze_these``` is now an argument for BaseTrainer. It accepts a list or tuple of strings. The strings are the names of models or loss functions, and these models or losses will be set to eval() and their optimizers will not be stepped.
